docs/source/reference/tpu.rst

```diff
@@ -37,12 +38,72 @@ Like :ref:`GPUs <interactive-nodes>`, SkyPilot provides a simple command to quic
 
 After the command has finished, you will be dropped into the host VM and can start develop code right away!
 
-Below we demonstrate how to run MNIST training on both TPU Nodes and TPU VMs with SkyPilot YAML.
+Below we demonstrate how to run MNIST training on both TPU VMs and TPU Nodes with SkyPilot YAML.
+
+TPU VMs
+--------------------------------
+
+To use TPU VMs, user needs to specify the TPU type, :code:`tpu_vm: True` and the desired TPU runtime version in :code:`accelerator_args` shown below:
+
+.. code-block:: yaml
+
+   resources:
+      accelerators: tpu-v2-8
+      accelerator_args:
+         runtime_version: tpu-vm-base
+         tpu_vm: True
+
+
+Now we show an example of running `mnist training <https://cloud.google.com/tpu/docs/run-calculation-jax#running_jax_code_on_a_tpu_vm>`_ on TPU VM with JAX.
+
+.. code-block:: yaml
+
+   name: mnist-tpu-vm
+
+   resources:
+      accelerators: tpu-v2-8
+      accelerator_args:
+         runtime_version: tpu-vm-base
+         tpu_vm: True
+
+   setup: |
+      git clone https://github.com/google/flax.git
+
+      conda activate flax
+      if [ $? -eq 0 ]; then
+         echo 'conda env exists'
+      else
+         conda create -n flax python=3.8 -y
+         conda activate flax
+         # Make sure to install TPU related packages in a conda env to avoid package conflicts.
+         pip install "jax[tpu]>=0.2.16" -f https://storage.googleapis.com/jax-releases/libtpu_releases.html
+         pip install --upgrade clu
+         pip install -e flax
+      fi
+
+   run: |
+      conda activate flax
+      cd flax/examples/mnist
+      python3 main.py --workdir=/tmp/mnist \
+      --config=configs/default.py \
+      --config.learning_rate=0.05 \
+      --config.num_epochs=10
+
+A GCS bucket is not required as the TPU VM is physically linked to the TPU device, which can access data directly.
+You are expected to see the below outputs when the job finishes.
+
+.. code-block:: console
+
+   $ sky launch examples/tpu/tpuvm_mnist.yaml -c mycluster
+   ...
+   (mnist-tpu-vm pid=10155) I0823 07:49:25.468526 139641357117440 train.py:146] epoch:  9, train_loss: 0.0120, train_accuracy: 99.64, test_loss: 0.0278, test_accuracy: 99.02
+   (mnist-tpu-vm pid=10155) I0823 07:49:26.966874 139641357117440 train.py:146] epoch: 10, train_loss: 0.0095, train_accuracy: 99.73, test_loss: 0.0264, test_accuracy: 99.19
+
 
 TPU Nodes
 --------------------------------
 
-To use TPU Node, a host CPU VM needs to be created together with a TPU node and configured correctly to connect with each other.
+Different from TPU VM, a host CPU VM needs to be created together with a TPU node and configured correctly to connect with each other.
```

Comment:
```suggestion
In a TPU Node, a normal host CPU VM is created and connected to the TPU device. This is different from a TPU VM.
```

Actually, I don't  get the point about "needs to be created together with a TPU node and configured correctly to connect with each other." -- if you do it using `gcloud` manually or in the GCP console, the user doesn't need to perform this manual effort, right?

Reply:
What I meant is: user needs to ensure the host VM is created in the same zone as the TPU node and configure `TPU_NAME` env variable. Otherwise they won't be properly connected. But I see the confusion. Not sure how to better express it.

Reply:
How about we don't mention the complexity. Suggestion: replace L114 with
> In a TPU Node, a normal `n1` CPU VM is provisioned and connected to the TPU host/device.

Then add a new line before L115?

Reply:
Thanks I updated with slight modification.